app/services/story_service.go

```diff
@@ -531,6 +532,15 @@ func (s *StoryService) UpdateStoryStatusByUser(storyID int, status string) error
 		return types.ErrInvalidStatus
 	}
 
+	existingStoryInProgress, err := s.GetStoryByProjectIdAndStatus(projectID, constants.InProgress)
+	if err != nil {
+		s.logger.Error("Error fetching story", zap.Error(err))
+        return err
+	}
+	if existingStoryInProgress!= nil {
+        s.logger.Error("Another story is already in progress.")
+        return errors.New("another story is already in progress!")
```

Comment:
make custom error/reuse if existing similar custom error and handle as badrequest from client in controller